### PR TITLE
Support wireframes for 2D meshes

### LINKED ISSUE #5881: Wireframe for 2D meshes
## What problem does this solve or what need does it fill?

It could be useful to debug the mesh generation for 2D objects.

## What solution would you like?

Support the same interface as 3D.

## What alternative(s) have you considered?

Just draw the triangles yourself with bevy_prototype_lyon or some other vector framework.

## Additional context

I've been looking over the code of the wireframe plugin, and yolo replaced the Opaque3D with Transparent2d but that just crashes with a long error. My knowledge of the rendering framework is very limited so don't know what else to try.

### PR DESCRIPTION
**Pull request by msvbg**
# Objective

Wireframes are currently supported for 3D meshes using the `WireframePlugin` in `bevy_pbr`. This PR adds the same functionality for 2D meshes.

Closes #5881.

## Solution

Since there's no easy way to share material implementations between 2D, 3D, and UI, this is mostly a straight copy and rename from the original plugin into `bevy_sprite`.

image

---

## Changelog

- Added `Wireframe2dPlugin` and related types to support 2D wireframes.
- Added an example to demonstrate how to use 2D wireframes

### CODE REVIEW

File: crates/bevy_sprite/src/mesh2d/wireframe2d.wgsl
Comment (IceSentry) [inline crates/bevy_sprite/src/mesh2d/wireframe2d.wgsl:8]:
```suggestion
@group(2) @binding(0) var<uniform> material: WireframeMaterial;
```

Tiny nitpick, the 3d wireframe shader doesn't do that apparently but we now use this convention for all internal shaders.

File: examples/2d/wireframe_2d.rs
Comment (IceSentry) [inline examples/2d/wireframe_2d.rs:125]:
```suggestion
C - Change color of the green circle wireframe
```

Comment (msvbg) [reply examples/2d/wireframe_2d.rs:125]: Thanks!

**Comment by IceSentry** [comment]
The example looks very different for me. I'm only seeing lines around the shapes but no inside triangles are visible for me

**Comment by IceSentry** [comment]
Also, can you rebase on main? I think the Color changes will break some stuff

**Comment by msvbg** [comment]
Not entirely sure why the `check-bans` job is failing, perhaps because I modified `Cargo.toml`? It seems to be complaining about duplicate lock file entries.

**Comment by alice-i-cecile** [comment]
Yep, the `check-bans` failure isn't your fault. It's a real-but-noisy failure that we're blocked on fixing due to version synchronization dances.

**Review by IceSentry — APPROVED** [review]

**Comment by msvbg** [comment]
I see that @IceSentry approved already, but I felt that I should use the new color types now that #12137 was merged. I added 80aea41b9a4fc7f05049758d6a2e82eee98f1d34 and 2d177ffc56dace94ce7fc2705b842fc5e58be69c.

**Comment by IceSentry** [comment]
I think adding it to bevy_internal should be done in a separate PR. I know it's tiny, but it makes it easier to keep track of history if it isn't mixed with unrelated things. I'm not entirely sure what the plan is for migration of bevy_color though.

**Comment by alice-i-cecile** [comment]
> I think adding it to bevy_internal should be done in a separate PR. I know it's tiny, but it makes it easier to keep track of history if it isn't mixed with unrelated things. I'm not entirely sure what the plan is for migration of bevy_color though.

Agreed. I'd like a small PR adding it to `bevy_internal` and exporting a `prelude` like the other crates. Feel free to open it @msvbg, otherwise I'll get to it momentarily.

**Comment by msvbg** [comment]
Cool, I'll update this PR once it's merged.

**Comment by SludgePhD** [comment]
This should fix https://github.com/bevyengine/bevy/issues/5881

**Comment by msvbg** [comment]
Oh, didn't realize we had two of these PRs. My bad for not searching properly.

**Review by alice-i-cecile — APPROVED** [review]
Looks sensible, and an important feature.

**Comment by alice-i-cecile** [comment]
@msvbg looks like there's been breaking changes on `main` that need to be adapted to.

**Comment by msvbg** [comment]
I'll get it fixed up.

**Comment by msvbg** [comment]
@alice-i-cecile I resolved the problems in 7451663 and checked that the example still looks good :)

**Comment by alice-i-cecile** [comment]
Thank you to everyone involved with the authoring or reviewing of this PR! This work is relatively important and needs release notes! Head over to https://github.com/bevyengine/bevy-website/issues/1292 if you'd like to help out.

**Pull request merged**
